fastplotlib/layouts/_figure.py

```diff
@@ -97,45 +100,75 @@ def __init__(
             subplot names
         """
 
+        if isinstance(shape, list):
+            raise NotImplementedError("bounding boxes for shape not yet implemented")
+            if not all(isinstance(v, (tuple, list)) for v in shape):
+                raise TypeError(
+                    "shape argument must be a list of bounding boxes or a tuple[n_rows, n_cols]"
+                )
+            for item in shape:
+                if not all(isinstance(v, (int, np.integer)) for v in item):
+                    raise TypeError(
+                        "shape argument must be a list of bounding boxes or a tuple[n_rows, n_cols]"
+                    )
+            self._mode: str = "rect"
```

Comment:
code comment here and below to indicate what `self._mode` is for (rect vs. grid)